### PR TITLE
Full datagen CI

### LINKED ISSUE #1698: Add continuous integration for non-testdata data generation
#1353 added `scx.toml` to testdata, but it's not in the current [`icuexportdata_uprops_full.zip`](https://github.com/unicode-org/icu/releases/download/release-70-1/icuexportdata_uprops_full.zip), so data generation with `--all-keys` has been broken since then, including at the `0.5` cut.

### PR DESCRIPTION
**Pull request by robertbastian**
Fixes #1698

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -136,6 +136,49 @@ jobs:
...
+        path: data/source
+        key: source-data-41-71
+    - name: Download CLDR source data
+      if: steps.source-data-cache.outputs.cache-hit != 'true'
+      uses: robinraju/release-downloader@v1.3
+      with: 
+        repository: "unicode-org/cldr-json"
+        tag: "41.0.0"
+        fileName: "cldr-41.0.0-json-full.zip"
+        out-file-path: "data/source/"
```
Comment (sffc) [inline .github/workflows/build-test.yml:159]: Nit: Put the data in a path that is covered by `.gitignore` (or add this directory to `.gitignore`)

**Review by sffc — APPROVED** [review]
Nice!!

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -136,6 +136,49 @@ jobs:
...
+      with: 
+        repository: "unicode-org/icu"
+        tag: "release-71-1"
+        fileName: "icuexportdata_uprops_full.zip"
+        out-file-path: "data/source/"
+    - name: Run datagen
+      uses: actions-rs/cargo@v1.0.1
+      with:
+        command: run
+        args: >
```
Comment (Manishearth) [inline .github/workflows/build-test.yml:172]: nit: cargo make command perhaps?

**Review by Manishearth — APPROVED** [review]

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -136,6 +136,49 @@ jobs:
         command: make
         args: ci-job-test-docs
 
+  # Runs full datagen on prod data
+  full-datagen:
```
[the remark this answers is not in the document]
Comment (robertbastian) [reply .github/workflows/build-test.yml:140]: We have `cargo make full-data` which does the same thing but uses the built-in downloader instead

File: .github/workflows/build-test.yml (hunk 2 above)
Comment (robertbastian) [reply .github/workflows/build-test.yml:172]: I think this is much better, a reader won't have to chase down another workflow. A make command wouldn't be reusable anyway

File: .github/workflows/build-test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Manishearth) [reply .github/workflows/build-test.yml:140]: right, but we have a pattern of having a task `ci-job-foo` for every CI job so you don't have to go digging to find the right task, could you add an alias? Ideally one that behaves as closely to this job as possible

[the remark this answers is not in the document]
Comment (robertbastian) [reply .github/workflows/build-test.yml:140]: So rename `full-data` to `ci-job-full-datagen`?

[the remark this answers is not in the document]
Comment (Manishearth) [reply .github/workflows/build-test.yml:140]: I'd rather keep the full-data job around and _add_ a `ci-job-full-datagen` to the main Makefile.toml that simply depends on `full-data`. That's what we typically do, we keep the `ci-job-foo` tasks rather clean and just depending on other tasks with the actual meat of it.

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (robertbastian) [reply .github/workflows/build-test.yml:159]: Are these tests not hermetic?

**Review by Manishearth — APPROVED** [review]

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (sffc) [reply .github/workflows/build-test.yml:159]: I just find it messy for generated files to go into the non-ignored area of a git repo; I think of a git repo as something that is readonly except within gitignore-covered directories

**Review by sffc — DISMISSED** [review]

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (robertbastian) [reply .github/workflows/build-test.yml]: I'd rather place it outside the repo, but the cache action doesn't seem to support that. I don't think we should add this to `.gitignore` though as it never runs locally and on CI it gets cleaned up immediately

**Review by Manishearth — APPROVED** [review]

**Review by sffc — COMMENTED** [review]
lgtm

**Pull request merged**